packages/eightballer/skills/contract_subscription/handlers.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2023 eightballer
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This package contains a scaffold of a handler."""
+
+from typing import Optional
+
+from aea.configurations.base import PublicId
+from aea.protocols.base import Message
+from aea.skills.base import Handler
+
+from packages.fetchai.protocols.default.message import DefaultMessage
+
+
+class WebSocketHandler(Handler):
+    """This class scaffolds a handler."""
+
+    SUPPORTED_PROTOCOL = DefaultMessage.protocol_id
+
+    def setup(self) -> None:
+        """Implement the setup."""
+
+    def handle(self, message: Message) -> None:
+        """
+        Implement the reaction to an envelope.
+
+        :param message: the message
+        """
+        self.context.logger.info("Received message: {}".format(message))
```

Comment:
this is just a draft implementation I suppose, to be dealt with as we figure out more details about the incoming data stream (and start working on the behaviour)?

Reply:
We put the callback here to peform the parsing of the events, and the extraction of the logs.

We can then send these events to a data aggregation skill.

That is where we save it to the database or whatever.

Nice thing about this approach, is we can have these behaviour running async i the background to the main fsm, effectively ensuring that our main fsm has up to date data all the time.